File: insee-promotion-client/app/components/enum/TypeConstruction.js


export class TypeConstruction{

    constructor(type) {
        this.type = type;
    }

    static findByType(type) {
        switch(type) {
            case 1 : return NEXT_CONSTRUCTION;
            case 2 : return NOW_CONSTRUCTION;
            case 3 : return NOW_CONSTRUCTION_V2;
            default: return null;
        }
    }

    getType() {
        return this.type;
    }

    
}

export const NEXT_CONSTRUCTION = new TypeConstruction(1);
export const NOW_CONSTRUCTION = new TypeConstruction(2);
export const NOW_CONSTRUCTION_V2 = new TypeConstruction(3);

File: ZaloBotServer/src/main/java/com/vn/insee/zalobot/common/type/TypeConstruction.java
package com.vn.insee.zalobot.common.type;

public enum  TypeConstruction {
    NEXT_CONSTRUCTION(1), NOW_CONSTRUCTION(2), NOW_CONSTRUCTION_V2(3), CONSTRUCTION_LOYALTY(4);
    private int type;

    TypeConstruction(int type) {
        this.type = type;
    }

    public int getType() {
        return type;
    }

    public static TypeConstruction findByType(int type) {
        switch (type) {
            case 1 : return NEXT_CONSTRUCTION;
            case 2 : return NOW_CONSTRUCTION;
            case 3 : return NOW_CONSTRUCTION_V2;
            case 4 : return CONSTRUCTION_LOYALTY;
            default: return null;
        }
    }
}
